datasets/esa-worldcover/esa_worldcover.py

```diff
@@ -0,0 +1,105 @@
+import logging
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import List, Optional, Union
+
+import numpy as np
+import numpy.typing as npt
+import pystac
+import rasterio
+from shapely.geometry import shape
+from shapely.geometry.multipolygon import MultiPolygon
+from shapely.geometry.polygon import Polygon
+from stactools.core.utils.raster_footprint import RasterFootprint
+from stactools.esa_worldcover.stac import create_item
+
+from pctasks.core.models.task import WaitTaskResult
+from pctasks.core.storage import StorageFactory
+from pctasks.dataset.collection import Collection
+
+handler = logging.StreamHandler()
+handler.setFormatter(logging.Formatter("[%(levelname)s]:%(asctime)s: %(message)s"))
+handler.setLevel(logging.INFO)
+logger = logging.getLogger(__name__)
+logger.addHandler(handler)
+logger.setLevel(logging.INFO)
+
+
+# This subclass is a workaround for node out-of-memory failures when computing
+# the data mask (the worldcover tiles are large: 36,000x36,000 pixels). We
+# bypass the mask creation and use the mask offered by rasterio. Since rasterio
+# masks use a valid data value of 255 rather than 1, we also override the
+# `data_extent` method. We do not convert the rasterio mask valid data value of
+# 255 to 1, as the conversion logic (np.where()) is the same as that which
+# causes the out-of-memory failure.
+#
+# This subclass may eventually be pushed to the stactools package.
+class RioMaskRasterFootprint(RasterFootprint):
```

Comment:
Didn't this just get pushed down to the **stactools-package** and can be removed?

Reply:
Yep. I removed the class and updated `requirements.txt` to point to the latest stactools-package release.